src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -143,12 +147,13 @@ public bool TryLoadConfig(
             string json = _fileSystem.File.ReadAllText(path);
             bool configParsed = TryParseConfig(json, out RuntimeConfig, connectionString: _connectionString, replaceEnvVar: replaceEnvVar);
             TrySetupConfigFileWatcher();
-            config = RuntimeConfig;
-            if (RuntimeConfig is not null)
+
+            if (RuntimeConfig is not null && !string.IsNullOrEmpty(defaultDataSourceName))
```

Comment:
why "!string.IsNullOrEmpty(defaultDataSourceName)" ? aren't we intentionally setting default datasource name to empty string ?

Reply:
This is for overwriting the default data source name, which we don't want to do in the case of an empty string, but only when a specific defaultDataSourceName is provided. So for example in a hot-reload scenario we can pass along the previous name and it will overwrite here, but in the case that we have a name already, we don't want to overwrite with the empty string.

Reply:
this would be a good comment to include in the code.